src/backend/input.rs

```diff
@@ -216,6 +231,9 @@ pub trait InputBackend: Sized {
     /// Get current `InputConfig`
     fn input_config(&mut self) -> &mut Self::InputConfig;
 
+    /// Called to inform the Input backend about a new focused Output for a `Seat`
+    fn set_output_metadata(&mut self, seat: &Seat, output: &Output);
```

Comment:
Generics would make `InputHandler` not object-safe, which is problematic for the `InputBackend` (just try it, this was the only solution I did found).

I though this would be more extensible and gives a nice API for the backends (you could just implement `Input::Output` for the `GraphicsBackend`s specific output type). I also think we have an API hole here, currently it looks like every `GraphicsBackend` is meant to be exactly one output, which will not work with DRM. I have some idea for that, but that belongs into a seperate design issue.

> Also, the question of device giving absolute positions (such as touchscreens) and locked to one output, independently of pointer focus remains.

I would say this is something a concrete implementation should handle. Just don't call set_output_metadata more then once for that output in that case.